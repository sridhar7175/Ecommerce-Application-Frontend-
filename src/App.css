* {
  padding: 0px;
  margin: 0px;
}
h2 {
  font-family: "Alata", sans-serif;
}
/* --->Header Start-------> */
.scrolled {
  position: fixed;
  background-color: gray;
}
.header {
  /* background-color: #ed017f; */
}

.navbar-brand {
  font-family: "Quicksand", sans-serif;
}
.header .nav-item .nav-link {
  font-size: 18px;
  font-family: "Quicksand", sans-serif;
}

.nav-link :active {
  border-bottom: #000 !important;
}

.nav-item :hover {
  color: #ed017f !important;
}
.nav-link {
  color: #000 !important;
}
[type="button"]:not(:disabled),
[type="reset"]:not(:disabled),
[type="submit"]:not(:disabled),
button:not(:disabled) {
  cursor: pointer;
  outline: none;
}
.dropbtn {
  border: none;
  text-decoration: none !important;
  color: #000 !important;
  font-size: 18px;
  font-family: "Quicksand", sans-serif;
}

.dropdown {
  position: relative;
  display: inline-block;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f1f1f1;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.dropdown-content .admindrp {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
}
.heart {
  text-align: right !important;
  margin-right: 10px;
  padding-top: 10px;
}

.dropdown-content .admindrp:hover {
  background-color: #ddd;
}

.dropdown:hover .dropdown-content {
  display: block;
}

.dropdown:hover .dropbtn {
  color: #ed017f !important;
}

/* --->Header End-------> */
/* --->Home Start--> */
/* .home123 {
  position: fixed;
} */
.home1 {
  font-size: 20px;
  text-decoration: none !important;
  color: white !important;
  padding: 5px 15px 5px 15px;
  border: 2px solid whitesmoke;
  /* margin-top: 15px !important; */
}
.carousel-caption {
  padding-bottom: 170px !important;
}
/* --->Home End--> */

/* --->Register and Login Start  */
.form-control {
  width: 300px !important;
  margin: auto !important;
  border-radius: 10px !important;
  outline: none !important;
  font-family: "Varela Round", sans-serif;
}
.register-box {
  width: 400px !important;
  height: 450px;
  font-weight: 400;
  background-color: #fff;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
  padding-top: 10px;
  border-radius: 10px;
}
.register-box h5 {
  font-family: "Kumbh Sans", sans-serif !important;
}
.login-box h5 {
  font-family: "Kumbh Sans", sans-serif !important;
}
.register-box p {
  font-family: "Nanum Gothic", sans-serif;
}
.login-box p {
  font-family: "Nanum Gothic", sans-serif;
}
.login-box {
  width: 400px !important;
  height: 350px;
  font-weight: 400;
  background-color: #fff;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
  padding-top: 10px;
  border-radius: 10px;
}
/* --->Register and Login End  */

/* ---->AboutUs Start----> */

/* --->AboutUs End----> */
.userdashboard1 .nav-link {
  font-family: "Quicksand", sans-serif;
}

.card1 {
  transition: 0.3s;
  border-radius: 10px !important;
  width: 250px !important;
  height: 400px;
  font-weight: 400;
  background-color: #fff;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
  /* padding-top: 20px; */
  background-clip: border-box;
  margin-bottom: 50px;
  margin-top: 20px;
}
.product123 {
  display: flex !important;
}
.product124 {
  text-align: right !important;
  margin-left: 650px;
}

.productbutton {
  border: 2px solid black;
  background-color: white;
  color: #000;
  padding: 5px 15px 5px 15px;
  font-size: 12px;
  font-weight: bold;
  cursor: pointer;
  border-radius: 3px;
}

/* Green */
.productbutton {
  border-color: gray;
  color: black;
}

.productbutton:hover {
  background-color: gray;
  color: white;
  transition: 0.5s;
}
.bg-info,
.btn-info {
  background: linear-gradient(45deg, #000, #add8e6);
}
.btn-info {
  background-color: #33b5e5 !important;
  color: #fff;
}
/* ---->About Us Start-----> */
.about3 {
  width: 100% !important;
  padding-top: 10px;
  height: 500px;
  font-weight: 400;
  background-color: #fff;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
  border-radius: 10px;
}
/* ---->About Us End-----> */

/*-> Footer Start---> */
.footer {
  background-color: #727272 !important;
  padding-top: 20px;
  width: 100%;
  height: 300px;
}
.footer h6 {
  font-family: "Overpass", sans-serif;
}
.footer .footer1 {
  border-top: 3px solid black !important;
  width: 30px !important;
}
.footer .footer1 .footer2 :hover {
  color: blue !important;
}
.footer .footer1 .footer3 :hover {
  color: blue !important;
}
.footer .footer1 .footer4 :hover {
  color: blue !important;
}

.footer6 {
  text-decoration: none !important;
  font-family: "Quicksand", sans-serif;
}

.form-control1 {
  width: 70% !important;
  border-radius: 20px;
  outline: none !important;
  border: none !important;
  font-family: "Quicksand", sans-serif;
}
.footer .btn {
  background-color: #2b2b52;
  border-radius: 3px;
  font-family: "Quicksand", sans-serif;
  font-weight: bold;
  font-size: 12px;
}
.footer p {
  font-family: "Quicksand", sans-serif;
}
.footer-form {
  width: 70%;
  box-sizing: border-box;
  /* border: 3px solid #ccc; */
  border: none !important;
  /* -webkit-transition: 0.5s; */
  /* transition: 0.5s; */
  outline: none;
  border-radius: 3px;
  padding: 5px 10px;
}

/* .footer-form:focus {
  border: 3px solid #555 !important;
} */
/* -->Footer End------> */

/* --->Admin Dashboard Start----> */
.newad1 {
  background-color: #333945 !important;
  width: 300px !important;
  height: 400px;
  border-radius: 10px;
  font-family: "PT Sans", sans-serif;
}
.newad2 li {
  list-style-type: none !important;
}
.newad3 {
  text-decoration: none !important;
  color: white !important;
}
.newad4 {
  background-color: #eaf0f1;
  width: 100% !important;
}
.admin-card {
  background-color: #43be31 !important;
  width: 300px;
  height: 200px;
  border-radius: 10px;
}
.admin-card1 {
  background-color: #ec4849;
  width: 300px;
  height: 200px;
  border-radius: 10px;
}
.admin-card2 {
  background-color: #4834df;
  width: 300px;
  height: 200px;
  border-radius: 10px;
}

/* --->Admin Dashboard End */

/* --->Slider Start----> */
/* -->Basic */
.react-multiple-carousel__arrow {
  outline: none !important;
}

.ourcoursescard {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  background-color: black !important;
  color: white;
  width: 350px;
  height: 350px;
  border-radius: 10px;
}
.react-multiple-carousel__arrow--left {
  left: calc(0% + -41px);
  top: 85px;
}

.react-multiple-carousel__arrow--right {
  right: calc(4% + 110px) !important;
  top: 85px !important;
}

/* -->Slider End----> */

/* .landinghome {
  background-color: #f0f2f5;
  width: 100%;
  height: 400px;
} */

/* -->Cart Start-------> */

.addtocart1 {
  border-bottom: 2px solid black !important;
}
.cart9 {
  padding-top: 10px !important;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  width: 90%;
  border-radius: 10px;
  margin-top: 90px !important;
  margin-left: 200px;
}
.cart9 btn-success {
  text-align: center !important;
}
.addtocart2 {
  margin-left: 1px;
}
/* 
.cart1 {
  font-family: "Quicksand", sans-serif;
}
.cart1 p {
  font-family: "Overpass", sans-serif;
}

.cart2 {
  border-bottom: 1px solid black;
  display: flex;
}
.cart3 {
  padding-left: 641px !important;
}
.cart4 {
  padding-left: 400px !important;
  margin-top: -61px;
}
.cart5 {
  border-bottom: 1px solid black;
}
.cart6 {
  font-weight: bold;
  padding-left: 400px !important;
}
.cart7 {
  border-bottom: 1px solid black;
}
.cart8 {
  padding-left: 650px !important;
  font-weight: bold;
}
.cart9 {
  padding-top: 10px !important;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  width: 90%;
  border-radius: 10px;
  margin-top: 50px !important;
}
.cart9 btn-success {
  text-align: center !important;
}

/* .card:hover {
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
} */
*/
/* -->Cart End-------> */

/* --->Product Details Start------> */

.pd1 h5 {
  font-family: "Quicksand", sans-serif;
  /* color: #926E4D; */
}
.pd1 p {
  font-family: "Quicksand", sans-serif;
}
.pd2 img {
  border-radius: 20px;
}
.pd3 h4 {
  font-family: "Quicksand", sans-serif;
}
.pd4 h5 {
  font-family: "Epilogue", sans-serif;
}
.pd4 li {
  font-family: "Zilla Slab", serif;
  font-size: 18px;
}
/* -->Product Details End-----> */

/* ---> Account Start---->*/
.ac1 {
  background-color: #f5f5f5;
  width: 100%;
  height: 40px;
  padding-left: 160px;
  padding-top: 8px;
  font-size: 15px;
  font-family: "Epilogue", sans-serif;
}
.ac1 .ac2 {
  text-decoration: none !important;
  color: #000;
}
.ac3 {
  border: 2px solid red;
  height: 300px !important;
}
.ac4 {
  border-bottom: 1px solid black;
  width: 200px;
}
.ac5 li {
  list-style-type: none !important;
}
.ac6 {
  text-decoration: none !important;
  color: black !important;
  font-family: "Peddana", serif;
}
.ac5 li :hover {
  color: #ff362e !important;
}
.ac7 {
  border: 2px solid red !important;
  height: 500px;
}
.form1 {
  width: 70%;
  padding: 5px 5px;
  box-sizing: border-box;
  border: 3px solid #ccc;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  outline: none;
}
.form1:focus {
  border: 3px solid #555;
}
.btn3 {
  background-color: #e71c23;
  color: white;
  border: none;
  padding: 10px 15px 10px 15px;
  border-radius: 10px;
  outline: none !important;
}
/* -->Account End */

/* --->WishList Start----> */
.wishlist1 {
  background-color: whitesmoke !important;
  width: 600px !important;
  height: 40px;
}
/* ---->WishList End---> */

/* -->shippingd1---> */
.shippingd1 {
  border: 2px solid red !important;
  height: 500px;
}
.shippingd3 {
  width: 200px !important;
}
.shippingd3 ::placeholder {
  font-size: 10px !important;
}
.btn4 {
  background-color: #e71c23;
  color: white;
  border: none;
  padding: 5px 15px 5px 15px;
  border-radius: 10px;
  outline: none !important;
}
.shopaddress {
  transition: 0.5s !important;
}
/* --->Account End-----> */
/* ---->User Dashboard-------> */
.ud1 {
  background-color: #f7f7f7;
  width: 100%;
  height: 70px;
  padding-left: 200px;
}
.ud1 h5,
h6 {
  font-family: "Assistant", sans-serif;
}
.ud2 {
  margin-left: 50px;
  background-color: #fff;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
}
.ud2 li {
  list-style-type: none;
  color: #000 !important;
}
.ud3 {
  text-decoration: none !important;
  font-size: 15px;
  color: #000 !important;
  font-family: "Abel", sans-serif;
  text-transform: capitalize;
}
.ud4 {
  border-bottom: 1px solid black !important;
  width: 349px;
}
.ud5 {
  border-right: 2px solid brown !important;
  height: 350px !important;
}
.ud6 {
  font-family: "Epilogue", sans-serif;
}
.btn1 {
  border: 2px solid black;
  background-color: white;
  color: black;
  padding: 14px 28px;
  font-size: 16px;
  cursor: pointer;
}

/* Green */
.success {
  border-color: rebeccapurple;
  color: black;
}

.success:hover {
  background-color: black;
  color: white;
}
.user-form1 .user-form2 {
  width: 500px !important;
}

/* ---->Media Quireies Start-------> */
@media only screen and (max-width: 600px) {
  .navbar-toggler {
    background-color: white !important;
    outline: none !important;
  }
  .card1 {
    margin: auto !important;
    text-align: center !important;
    margin-bottom: 20px !important;
  }
  .footer {
    background-color: #727272 !important;
    padding-top: 20px;
    height: 550px;
    width: 120% !important;
  }
  .register-box {
    width: 300px !important;
    height: 500px;
    font-weight: 400;
    background-color: #fff;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16),
      0 2px 10px 0 rgba(0, 0, 0, 0.12);
    padding-top: 10px;
    border-radius: 10px;
  }
  .login-box {
    width: 300px !important;
    height: 300px;
    font-weight: 400;
    background-color: #fff;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16),
      0 2px 10px 0 rgba(0, 0, 0, 0.12);
    padding-top: 10px;
    border-radius: 10px;
  }
  .form-control {
    width: 250px !important;
    margin: auto !important;
    border-radius: 10px !important;
    outline: none !important;
  }
  .about3 {
    width: 100% !important;
    padding-top: 10px;
    height: 1000px;
    font-weight: 400;
    background-color: #fff;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16),
      0 2px 10px 0 rgba(0, 0, 0, 0.12);
    border-radius: 10px;
  }

  .cart9 {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;
    width: 90%;
    border-radius: 10px;
    margin-top: 10px !important;
    margin-bottom: 30px;
    margin-left: 10px !important;
  }
  .addtocart2 {
    margin-left: -60px;
  }
  .addtocart3 {
    padding-left: 50px !important;
  }
  .addtocart1 {
    border-bottom: 2px solid black !important;
    width: 350px !important;
  }
}
@media screen and (max-width: 992px) {
  .navbar-toggler {
    background-color: white !important;
    outline: none !important;
  }
  .card1 {
    text-align: center !important;
  }
  .footer {
    background-color: #727272 !important;
    padding-top: 20px;
    width: 100%;
    height: 520px;
  }

  .cart3 {
    padding-left: 170px !important;
  }
  .cart4 {
    padding-left: 170px !important;
    margin-top: -20px !important;
  }
  .cart6 {
    font-weight: bold;
    padding-left: 20px !important;
  }
  .cart2 {
    border-bottom: 1px solid black;
    display: flex;
    width: 100% !important;
  }
  .cart8 {
    padding-left: 70px !important;
    font-weight: bold;
  }
  .cart9 {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;
    width: 90%;
    border-radius: 10px;
    margin-top: 10px !important;
    margin-bottom: 30px;
  }
  .ud1 {
    background-color: #f7f7f7;
    width: 100%;
    height: 70px;
    padding-left: 70px;
    font-family: "Epilogue", sans-serif;
  }
  .ud2 {
    margin-left: 10px;
  }

  .ud4 {
    border-bottom: 1px solid black !important;
    width: 250px;
  }
  .ud5 {
    border-right: 2px solid brown !important;
    width: 250px;
  }
  .product124 {
    text-align: right !important;
    margin-left: 100px;
  }
  .ac3 {
    margin-left: 10px;
    border: 2px solid red;
    width: 350px !important;
    height: 300px !important;
  }
  .ac1 {
    background-color: #f5f5f5;
    width: 100%;
    padding-left: 5px;
  }
}
/* ---->Media Quireies End-------> */
