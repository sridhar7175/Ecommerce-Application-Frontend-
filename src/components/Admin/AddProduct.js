import React, { useState, useEffect } from "react";
import Axios from "axios";
import { Link } from "react-router-dom";
import Swal from "sweetalert2";
import { RiAdminFill } from "react-icons/ri";
import { FiBox } from "react-icons/fi";
import { AiOutlineMedicineBox } from "react-icons/ai";
import { CgToolbox } from "react-icons/cg";
import { AiOutlineUsergroupAdd } from "react-icons/ai";
import { FaHome, FaFirstOrderAlt } from "react-icons/fa";

const AddProduct = (props) => {
  const [productName, setProductName] = useState("");
  const [productBrand, setProductBrand] = useState("");
  const [productDescription, setProductDescription] = useState("");
  const [productPrice, setProductPrice] = useState("");
  const [productPicture, setProductPicture] = useState("");

  const onChangeProduct = (e) => {
    setProductName(e.target.value);
  };

  useEffect(() => {
    console.log(productPicture, "productPicture");
  }, [productPicture]);

  const handlePhoto = (e) => {
    let reader = new FileReader();
    let file = e.target.files[0];
    reader.onloadend = () => {
      setProductPicture(file);
    };

    reader.readAsDataURL(file);
  };

  const addproduct = (e) => {
    e.preventDefault();
    console.log(productPicture, "pic");
    const formData = new FormData();
    formData.append("productPicture", productPicture);
    formData.append("productName", productName);
    formData.append("productBrand", productBrand);
    formData.append("productDescription", productDescription);
    formData.append("productPrice", productPrice);

    Axios.post("http://localhost:5000/api/createproduct", formData, {})
      .then((response) => {
        console.log(response);

        Swal.fire({ title: "created successfully", timer: 1500 });
      })
      .catch((err) => {
        console.log(err);
        Swal.fire({ title: "Please Enter All Details", timer: 1500 });
      });
    setProductName("");
    setProductBrand("");
    setProductPrice("");
    setProductDescription("");
    setProductPicture("");
  };

  return (
    <div>
      <div className="container mb-5 mt-5">
        <div className="row">
          <div className="col-sm-4">
            <div className="newad1 pl-3 pt-4">
              <img
                src="https://secure.gravatar.com/avatar/053467956553007b62a4cc9eb47ad16d?s=70&d=mm&r=g"
                alt="aa"
                style={{ borderRadius: "10px" }}
              />
              <h5 className="mt-2 text-white">Admin Dashboard</h5>
              <div className="newad2">
                <ul className="mt-2">
                  <li className="mt-1">
                    <Link className="newad3 mt-2" to="/admin/dashboard">
                      <FaHome /> Dashboard
                    </Link>
                  </li>
                  <li className="mt-1">
                    <Link className="newad3 mt-2" to="/admin/create/product">
                      <FiBox /> Create Product
                    </Link>
                  </li>
                  <li className="mt-1">
                    <Link className="newad3 mt-2" to="/admin/mangeproduct">
                      <FiBox /> Manageproduct
                    </Link>
                  </li>
                  {/* <li className="mt-1">
                    <Link className="newad3 mt-2" to="/admin/create/category">
                      <AiOutlineMedicineBox /> Createcategory
                    </Link>
                  </li>
                  <li className="mt-1">
                    <Link className="newad3 mt-2" to="/admin/managecategory">
                      <AiOutlineMedicineBox /> Managecategory
                    </Link>
                  </li>
  */}
                  <li className="mt-1">
                    <Link className="newad3 mt-2" to="/admin/Orders">
                      <FaFirstOrderAlt /> Orders
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div className="col-sm-8">
            <div
              className="container p-3  mb-5 pb-5 bg-info text-white"
              style={{ borderRadius: "10px" }}
            >
              <Link className="btn btn-warning" to={`/admin/dashboard`}>
                <span className="">Admin Home</span>
              </Link>
              <h1
                className="pt-3 text-center"
                style={{ fontFamily: "Alata, sans-serif" }}
              >
                Add New Products here!
              </h1>
              <div className="box3 rounded p-5">
                <form>
                  {productPicture && productPicture.length}
                  <div>
                    <input
                      className="form-control mt-2"
                      onChange={(e) => handlePhoto(e)}
                      type="file"
                      placeholder="choose a file"
                    />
                  </div>
                  <div>
                    <input
                      type="text"
                      name="productName"
                      className="form-control mt-2"
                      value={productName}
                      placeholder="ProductName..."
                      onChange={onChangeProduct}
                    />
                  </div>
                  <div>
                    <input
                      type="text"
                      name="productBrand"
                      className="form-control mt-2"
                      value={productBrand}
                      placeholder="ProductBrand..."
                      onChange={(e) => setProductBrand(e.target.value)}
                    />
                  </div>
                  <div>
                    <input
                      type="text"
                      name="productDescription"
                      className="form-control mt-2"
                      value={productDescription}
                      placeholder="productDescription..."
                      onChange={(e) => setProductDescription(e.target.value)}
                    />
                  </div>
                  <div>
                    <input
                      type="text"
                      name="productPrice"
                      className="form-control mt-2"
                      value={productPrice}
                      placeholder="ProductPrice..."
                      onChange={(e) => setProductPrice(e.target.value)}
                    />
                  </div>

                  <button
                    onClick={addproduct}
                    className="mt-3 mb-4 form-control btn-success"
                  >
                    CreateProduct
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AddProduct;
