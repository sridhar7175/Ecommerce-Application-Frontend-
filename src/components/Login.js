import React, { useState } from "react";
import Axios from "axios";
import { connect } from "react-redux";
const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const login = (e) => {
    e.preventDefault();
    var body = { email: email, password: password };

    Axios.post("http://localhost:5000/api/user/signin", body)
      .then((res) => {
        console.log(res.data);
        localStorage.setItem("userId", res.data._id);
        localStorage.setItem("role", res.data.role);
        localStorage.setItem("name", res.data.name);
        localStorage.setItem("email", res.data.email);
        window.location.href = "/";
      })
      .catch((err) => {
        console.log(JSON.stringify(err));
        alert("Failed to Login, Please try again");
      });
    setEmail("");
    setPassword("");
  };
  return (
    <div className="text-center container mt-5">
      <h1>Sign In page</h1>
      <h4 className="mt-2 mb-2">A page for user to sign in!</h4>
      <form>
        <div>
          <input
            type="text"
            name="email"
            className="form-control mt-3"
            placeholder="Enter a Email....."
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </div>
        <div>
          <input
            type="text"
            name="password"
            className="form-control mt-3"
            placeholder="Enter a Password....."
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <button onClick={login} className="form-control btn-danger mt-3">
          Submit
        </button>
      </form>
    </div>
  );
};
const mapStateToProps = (state) => {
  return {};
};

const mapDispatchToProps = (dispatch) => {
  return {};
};

export default connect(mapStateToProps,mapDispatchToProps)(Login);
