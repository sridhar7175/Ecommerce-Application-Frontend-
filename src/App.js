import React from "react";
import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Homepage from "./containers/Homepage";
import Registerpage from "./containers/Registerpage";
import Loginpage from "./containers/Loginpage";
import Cartpage from "./containers/Cartpage";
import Addproductpage from "./containers/adminpages/Addproductpage";
import Buypage from "./containers/buypages/Buypage";
import Manageproductspage from "./containers/adminpages/Manageproductspage";
import Updateproductpage from "./containers/adminpages/Updateproductpage";
import Shoppage from "./containers/commonpages/Shoppage";
import Productdetailspage from "./containers/adminpages/Productdetailspage";
import AccountInfopage from "./containers/accountpages.js/AccountInfopage";
import ShippingDetailspage from "./containers/accountpages.js/ShippingDetailspage";
import Wishlistpage from "./containers/accountpages.js/Wishlistpage";
import Orderspage from "./containers/accountpages.js/Orderspage";
import AdminDashboardpage from "./containers/adminpages/AdminDashboardpage";
import Addcategorypage from "./containers/adminpages/Addcategorypage";
import ManageCategorypage from "./containers/adminpages/ManageCategorypage";
import Adminorderpage from "./containers/adminpages/Adminorderpage";
import LandingPage from "./containers/LandingPage";

function App(props) {
  return (
    <div className="">
      <Router>
        <Switch>
          <Route exact path="/" component={Homepage} />
          {/*  <Route path="/home" component={Homepage} />*/}
          <Route path="/shop" component={Shoppage} />
          <Route path="/signup" component={Registerpage} />
          <Route path="/signin" component={Loginpage} />
          <Route path="/cart" component={Cartpage} />
          <Route path="/accountinfo" component={AccountInfopage} />
          <Route path="/shippingDetails" component={ShippingDetailspage} />
          <Route path="/wishlist" component={Wishlistpage} />
          <Route path="/myorder" component={Orderspage} />
          <Route path="/admin/dashboard" component={AdminDashboardpage} />
          <Route path="/admin/create/product" component={Addproductpage} />
          <Route path="/admin/mangeproduct" component={Manageproductspage} />
          <Route path="/admin/create/category" component={Addcategorypage} />
          <Route path="/admin/managecategory" component={ManageCategorypage} />
          <Route path="/admin/Orders" component={Adminorderpage} />
          <Route
            path="/admin/product/update/:Id"
            component={Updateproductpage}
          />
          <Route path="/buy" component={Buypage} />
          <Route path="/productdetails/:Id" component={Productdetailspage} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
