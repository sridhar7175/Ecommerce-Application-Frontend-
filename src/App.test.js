import React from "react";

import App from "./App";

import renderer from "react-test-renderer";
import Register from "./components/Register";

import Login from "./components/Login";
import { render, fireEvent } from "@testing-library/react";
import AddProduct from "./components/Admin/AddProduct";

describe("<Register/>", () => {
  it("It renders Correctly", () => {
    const tree = renderer.create(<Register />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("It renders 2 Correctly", () => {
    const handleClick = jest.fn();
    const { container } = render(<Register />);
    fireEvent.click(container.querySelector("button"));
    expect(handleClick).toBeCalledTimes(0);
  });
});

// describe("<Login/>", () => {
//   it("it Renders Login Correctly", () => {
//     const tree = renderer.create(<Login />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//   it("it renders 2 Login Correctly", () => {
//     const handleClick = jest.fn();
//     const { container } = render(<Login />);
//     fireEvent.click(container.querySelector("button"));
//     expect(handleClick).toBeCalledTimes(0);
//   });
// });

// describe("<AddProduct/>", () => {
//   it("it renders addproduct correctly", () => {
//     const tree = renderer.create(<AddProduct />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//   it("it renders 2 correctly add", () => {
//     const result = jest.fn();
//     const { container } = render(<AddProduct />);
//     fireEvent.click(container.querySelector("button"));
//     expect(result).toBeCalledTimes(0);
//   });
// });
