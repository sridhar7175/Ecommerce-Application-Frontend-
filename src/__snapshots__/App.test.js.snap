// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> It renders Correctly 1`] = `
<div
  className=""
>
  <div
    className="container mt-5 register-box mb-5"
  >
    <h5
      className="pl-5 mt-2"
    >
      Signup Here
    </h5>
    <form>
      <div>
        <input
          className="form-control mt-3"
          name="name"
          onChange={[Function]}
          placeholder="Enter a name....."
          type="text"
          value=""
        />
      </div>
      <div
        className="pl-5"
        style={
          Object {
            "color": "red",
            "fontSize": 10,
            "fontWeight": "bold",
            "outlineColor": "red",
          }
        }
      >
        
      </div>
      <div>
        <input
          className="form-control mt-3"
          name="email"
          onChange={[Function]}
          placeholder="Enter a Email....."
          type="email"
          value=""
        />
      </div>
      <div
        className="pl-5"
        style={
          Object {
            "color": "red",
            "fontSize": 10,
            "fontWeight": "bold",
            "outlineColor": "red",
          }
        }
      >
        
      </div>
      <div>
        <select
          className="form-control mt-3"
          name="role"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Please Select
          </option>
          <option>
            user
          </option>
          <option>
            admin
          </option>
        </select>
      </div>
      <div
        className="pl-5"
        style={
          Object {
            "color": "red",
            "fontSize": 10,
            "fontWeight": "bold",
            "outlineColor": "red",
          }
        }
      >
        
      </div>
      <div>
        <input
          className="form-control mt-3"
          name="password"
          onChange={[Function]}
          placeholder="Enter a Password....."
          type="password"
          value=""
        />
      </div>
      <div
        className="pl-5"
        style={
          Object {
            "color": "red",
            "fontSize": 10,
            "fontWeight": "bold",
            "outlineColor": "red",
          }
        }
      >
        
      </div>
      <button
        className="form-control btn-success mt-3 mb-2"
        onClick={[Function]}
      >
        Submit
      </button>
      <p
        className="mt-3 pl-5"
        style={
          Object {
            "fontSize": "12px",
            "fontWeight": "bold",
          }
        }
      >
        Already have an account?
      </p>
    </form>
  </div>
</div>
`;
